### PR TITLE
chore: simplify dependencies

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

- move some strapi/strapi deps to strapi/core & cleanup some circular deps.

### Why is it needed?

To address some circular dependency issues. We still have some we cannot solve for a while (admin/strapi/core/admin)

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/content-type-builder/package.json [hunk 1]
```diff
@@ -77,7 +77,6 @@
   },
   "devDependencies": {
     "@strapi/pack-up": "4.15.5",
-    "@strapi/strapi": "4.15.5",
```
Comment (Convly) [inline packages/core/content-type-builder/package.json:80]: Don't we need it anymore? What has changed?

File: packages/core/upload/package.json [hunk 2]
```diff
@@ -71,7 +71,6 @@
   },
   "devDependencies": {
     "@strapi/pack-up": "4.15.5",
-    "@strapi/strapi": "4.15.5",
```
Comment (Convly) [inline packages/core/upload/package.json:74]: Same than for the CTB

File: packages/core/content-type-builder/package.json (hunk 1 above)
Comment (alexandrebodin) [reply packages/core/content-type-builder/package.json:80]: it'as actually never imported and the types are coming from another package

File: packages/core/upload/package.json (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/upload/package.json:74]: same explanation

File: packages/core/strapi/package.json [hunk 3]
```diff
@@ -112,10 +112,6 @@
     "@strapi/logger": "4.15.5",
     "@strapi/pack-up": "4.15.5",
     "@strapi/permissions": "4.15.5",
-    "@strapi/plugin-content-manager": "4.15.5",
-    "@strapi/plugin-content-type-builder": "4.15.5",
-    "@strapi/plugin-email": "4.15.5",
-    "@strapi/plugin-upload": "4.15.5",
```
Comment (joshuaellis) [inline packages/core/strapi/package.json:118]: I don't think these should be removed. They're imported as they're "core" plugins. https://github.com/strapi/strapi/blob/4ad8ac0235a95a1dfb3b9e2f33633b2fbd14ef71/packages/core/strapi/src/admin.ts

Comment (joshuaellis) [reply packages/core/strapi/package.json:118]: maybe they were meant to be removed from the `core` package...?

Comment (alexandrebodin) [reply packages/core/strapi/package.json:118]: true they should be only removed from core/core idk why it's removed here 🤔

**Review by Convly — APPROVED** [review]
LGTM

**Comment by alexandrebodin** [comment]
I'm going to directly merge https://github.com/strapi/strapi/pull/19181 as it contains both 👍  Thank you!

**Pull request closed without merging**